ui/apps/platform/src/Containers/NetworkGraph/NetworkGraphContainer.tsx

```diff
@@ -150,6 +155,39 @@ function getDisplayEdges(edges: CustomEdgeModel[], showEdgeLabels: boolean): Cus
     });
 }
 
+// This function modifies the nodes to add another data attribute to distinguish faded out nodes from normal ones
+function fadeOutUnconnectedNodes(
+    nodes: CustomNodeModel[],
+    edges: CustomEdgeModel[],
+    selectedNodeId: string | undefined
+) {
+    const connectedNodeIds = getConnectedNodeIds(nodes, edges, selectedNodeId);
+    const modifiedNodes: CustomNodeModel[] = nodes.map((node) => {
+        const { data } = node;
+        if (
+            data.type === 'DEPLOYMENT' ||
+            data.type === 'CIDR_BLOCK' ||
+            data.type === 'EXTERNAL_ENTITIES'
+        ) {
+            const isConnectedToSelectedNode = connectedNodeIds.includes(node.id);
+            const isSelectedNode = node.id === selectedNodeId;
+            const isNodeSelected = !!selectedNodeId;
+            const isFadedOut = isNodeSelected
+                ? !isConnectedToSelectedNode && !isSelectedNode
+                : false;
+            return {
+                ...node,
+                data: {
+                    ...data,
+                    isFadedOut,
+                },
+            } as CustomNodeModel;
```

Comment:
Could we short-circuit this sooner? It seems like if there is no `selectedNodeId` given, then there is no reason to calculate `isConnectedToSelectedNode` or `isSelectedNode`

Reply:
Actually, I think we want to keep it like this because we do want to update the node data correctly whenever we go from selecting a node to unselecting a node